Add ignoreFields doc (#466)
open-cluster-management-io/open-cluster-management-io.github.io: content/en/docs/concepts/work-distribution/manifestwork.md

Instead of create the second `ManifestWork`, user can also set HPA for this deployment. HPA will also take the ownership
of `replicas`, and the update of `replicas` field in the first `ManifestWork` will return conflict condition.

## Ignore fields in Server Side Apply

> To avoid work-agent retuning conflict error, when using ServerSideApply as the update strategy, users can specify certain

Reviewer: returning?

fields to be ignored, such that when work agent is applying the resource to the `ManagedCluster`, the change on the
specified fields will not be updated onto the resource.

It is useful when other actors on the `ManagedCluster` is updating the same field on the resources
that the `ManifestWork` is owning. One example as below:

```yaml
apiVersion: work.open-cluster-management.io/v1
kind: ManifestWork
metadata:
  namespace: <target managed cluster>
  name: hello-work-demo
spec:
  workload: ...
  manifestConfigs:
    - resourceIdentifier:
        resource: configmaps
        namespace: default
        name: some-configmap
      updateStrategy:
        type: ServerSideApply
        force: true
        serverSideApply:
          ignoreFields:
            - condition: OnSpokePresent
              jsonPaths:
                - .data
```

It indicates that when the configmap is applied on the `ManagedCluster`, any additional change

> on the data field will be ignored by the work agent, no matter the change comes from a another

Reviewer: ```suggestion
on the data field will be ignored by the work agent, no matter the change comes from an
```
